modules/csv/csv.go

```diff
@@ -7,16 +7,18 @@ package csv
 import (
 	"bytes"
 	stdcsv "encoding/csv"
-	"errors"
 	"io"
+	"path/filepath"
 	"regexp"
 	"strings"
 
+	"code.gitea.io/gitea/modules/markup"
 	"code.gitea.io/gitea/modules/translation"
 	"code.gitea.io/gitea/modules/util"
 )
 
-var quoteRegexp = regexp.MustCompile(`["'][\s\S]+?["']`)
+const maxLines = 10
+const guessSampleSize = 1e4 // 10k
```

Comment:
Do we count file size in decimal or binary units?
Because in binary, this number should be a bit larger (10192, or 2^13, if I'm not mistaken).
I know that oftentimes, compilers prefer powers of two as those can be optimized better. I guess that should be the case here as well?

Reply:
This isn't something I added, as it was always a sample size of 10,000. So yeah, it is using 'k' here in the comments that were there before I worked on this as decimal. It's just an arbitrary sample size.

Reply:
> It's just an arbitrary sample size

Yes. And I'm arguing to set it to a value that the compiler will be able to optimize better.

One small correction, though: Apparently, I thought that 2^(11+1) would be 5096, instead of 4096.
So, what I am asking for is to at best use either `1<<13` (`8192`) or `1<<14` (`16384`) bytes.
Which do you prefer?
Or do you prefer the `10000`, which is slightly easier to read, but will be worse in runtime performance?

Reply:
It doesn't matter whatever we choose, 1e4, 16384, 8192, all work, you won't feel any difference.

We are not writing an OS or a memory allocation library, we won't benefit from the page-aligned memory size.

We do not need to struggle on such trivial details.